next-app/app/page.js

```diff
@@ -1,30 +1,127 @@
+import { ListGroup, OverlayTrigger, Toast, ToastContainer } from 'react-bootstrap';
 import Button from 'react-bootstrap/Button';
 import Tooltip from 'react-bootstrap/Tooltip';
 import Error from './components/loading-and-error/error';
 import Loading from './components/loading-and-error/loading';
+import './mobile.css';
 import useRequest from './utils/useRequest';
 
 // Main component
 export default function Home() {
-  const { response, error, loading } = useRequest('GET', '/api/weather');
+  const {
+    response: weather,
+    error: weatherErr,
+    loading: weatherLoading,
+  } = useRequest('GET', '/api/weather');
+  const {
+    response: schedule,
+    error: scheduleErr,
+    loading: scheduleLoading,
+    refetch,
+  } = useRequest('GET', '/api/schedule');
+  const [showToast, setShowToast] = useState(false);
+  const [dateRequested, setDateRequested] = useState(null);
 
-  if (loading) {
+  function scheduleDupe(date) {
+    let duplicate = false;
+
+    schedule.forEach((schedule) => {
+      const { sdate } = schedule;
+      const formatSdate = new Date(sdate.slice(0, 10) + 'T00:00:00.000-05:00').toDateString();
+
+      if (formatSdate === date.toDateString()) {
+        duplicate = true;
+      }
+    });
+
+    return duplicate;
+  }
+
+  async function addDate(scheduleDate) {
+    const url = '/api/schedule';
+
+    if (scheduleDate === '') {
+      return;
+    }
+
+    if (scheduleDupe(scheduleDate)) {
+      setDateRequested('error');
+      return;
+    }
+
+    try {
+      const response = await fetch(url, {
+        method: 'POST',
+        body: JSON.stringify({ sdate: scheduleDate }),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      });
+
+      if (!response.ok) {
+        throw new Error(`Response status: ${response.status}`);
+      }
+
+      refetch();
+      setDateRequested('success');
+    } catch (error) {
+      console.error(error.message);
+    }
+  }
+
+  async function undo(undoDate) {
+    const url = '/api/schedule';
+    let id = null;
+
+    schedule.forEach((dates) => {
+      if (dates.sdate.slice(0, 10) === undoDate.slice(0, 10)) {
+        id = dates.id;
+      }
+    });
+
+    try {
+      const response = await fetch(url, {
+        method: 'DELETE',
+        body: JSON.stringify({ id }),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      });
+
+      if (!response.ok) {
+        throw new Error(`Response status: ${response.status}`);
+      }
+
+      refetch();
+    } catch (error) {
+      console.error(error.message);
+    }
+  }
+
+  if (weatherLoading || scheduleLoading) {
     return <Loading />;
   }
 
-  if (error) {
+  if (weatherErr || scheduleErr) {
     return <Error />;
   }
 
   return (
     <div>
-      <h1 className="display-1 text-center p-auto m-auto pt-2">PikoWeatherer</h1>
+      <h1 className="display-1 text-center p-auto m-auto pt-2 title">PikoWeatherer</h1>
       <br />
```

Comment:
i know these `br`s arent part of this PR, but generally, we should try to avoid them and us HTML/CSS margin/padding features